include/lsst/afw/typehandling/test.h

```diff
@@ -203,6 +204,155 @@ BOOST_TEST_CASE_TEMPLATE_FUNCTION(TestEquals, GenericMapFactory) {
     BOOST_CHECK(*map1 != *(factory.makeGenericMap()));
 }
 
+BOOST_TEST_CASE_TEMPLATE_FUNCTION(TestConstVisitor, GenericMapFactory) {
+    static GenericMapFactory const factory;
+    std::unique_ptr<GenericMap<int> const> map = factory.makeGenericMap();
+    std::vector<int> mapKeys = map->keys();
+
+    // Visitors compute sizeof because it's one of the few operations valid for an expression of *any* type
+    // This lets us test generic lambdas as visitors
+    auto bruteForceSizer = [](int key) {
+        switch (key) {
+            case 0:
+                return sizeof(decltype(KEY0)::ValueType);
+            case 1:
+                return sizeof(decltype(KEY1)::ValueType);
+            case 2:
+                return sizeof(decltype(KEY2)::ValueType);
+            case 3:
+                return sizeof(decltype(KEY3)::ValueType);
+            case 4:
+                return sizeof(decltype(KEY4)::ValueType);
+            case 5:
+                // sizeof considers only static type, so visitor won't return sizeof(ComplexStorable)
+                return sizeof(Storable);
+            default:
+                throw std::invalid_argument("Bad key found");
+        };
+    };
+    std::vector<size_t> expected;
+    for (int key : mapKeys) {
+        expected.push_back(bruteForceSizer(key));
+    }
+
+    // Test local class that returns void
+    class {
+    public:
+        size_t totalSize = 0u;
+
+        // Local classes can't have method templates
```

Comment:
In production code there would be much less reason to make it a local class; you could just stick a regular class with template methods (like the `Printer` example) in your source file. But because `test.h` is a file that gets included by the actual test cases, I was a little worried about defining top-level classes that are only needed by a single test case.